kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -96,6 +98,19 @@ func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs) (*KafkaDispat
 	}
 	receiverFunc, err := eventingchannels.NewMessageReceiver(
 		func(ctx context.Context, channel eventingchannels.ChannelReference, message binding.Message, transformers []binding.Transformer, _ nethttp.Header) error {
+
+			// If we record the trace, we need to write the span info into the event
+			// using the distributed tracing extension.
+			// If the span is not recorded, we just don't send these info
+			span := trace.FromContext(ctx)
+			defer span.End()
+			if span.IsRecordingEvents() {
```

Comment:
trace context propagation should probably happen regardless of whether events are recorded.

Reply:
I think what we do here is fine because it's similar to doing `span.IsRecordingEvents()`: if you're not recording the span, there is no need to do the hard lifting for writing the span and reading it from the other side.

Reply:
Trace context is meant to be propagated regardless of sampling decision, otherwise there would be no need for a sampled flag. Here's zipkin's documentation on the reason for doing so: https://github.com/openzipkin/b3-propagation#why-send-trace-ids-with-a-reject-sampling-decision.